bleak/backends/bumble/scanner.py

```diff
@@ -0,0 +1,124 @@
+            containing this advertising data will be received.
+    Keyword Args:
+        adapter (BumbleTransport): Bumble transport adapter to use.
+    """
+
+    def __init__(
+        self,
+        detection_callback: Optional[AdvertisementDataCallback],
+        service_uuids: Optional[List[str]],
+        scanning_mode: Literal["active", "passive"],
+        **kwargs,
+    ):
+        super(BumbleScanner, self).__init__(detection_callback, service_uuids)
+        self.device = Device("scanner_dev", address=Address(SCANNER_MAC_ADDR))
+
+        self.device.host = Host()
+        self.device.host.controller = Controller("Scanner", link=get_link())
+        self.device.listener = self
+        self._scan_active = scanning_mode == "active"
+        self._adapter: BumbleTransport = kwargs.get("adapter", get_default_transport())
+
+    def on_advertisement(self, advertisement: Advertisement):
+        service_uuids: List[str] = []
+        service_data: Dict[str, AdvertisingDataObject] = {}
+        local_name = advertisement.data.get(AdvertisingData.COMPLETE_LOCAL_NAME)
+        if not local_name:
+            local_name = advertisement.data.get(AdvertisingData.SHORTENED_LOCAL_NAME)
+        manuf_data = advertisement.data.get(AdvertisingData.MANUFACTURER_SPECIFIC_DATA)
+        for uuid_type in SERVICE_UUID_TYPES:
+            adv_uuids = advertisement.data.get(uuid_type)
+            if adv_uuids is None:
+                continue
+            if not isinstance(adv_uuids, list):
+                continue
+            for uuid in adv_uuids:
+                if uuid not in service_uuids:
+                    service_uuids.append(str(uuid))
+
+        for service_data in advertisement.data.get_all(AdvertisingData.SERVICE_DATA):
+            service_uuid, data = service_data
+            service_data[str(service_uuid)] = data
+
+        advertisement_data = AdvertisementData(
+            local_name=local_name,
+            manufacturer_data=manuf_data,
+            service_data=service_data,
+            service_uuids=service_uuids,
+            tx_power=advertisement.tx_power,
+            rssi=advertisement.rssi,
+            platform_data=(None, None),
+        )
+
+        device = self.create_or_update_device(
+            str(advertisement.address),
+            local_name,
+            {},
+            advertisement_data,
+        )
+
+        self.call_detection_callbacks(device, advertisement_data)
+
+    async def on_connection(self, connection):
+        pass
+
+    async def start(self) -> None:
+        await start_transport(self._adapter)
+        await self.device.power_on()
+        await self.device.start_scanning(active=self._scan_active)
+
+    async def stop(self) -> None:
+        await self.device.stop_scanning()
+        await self.device.power_off()
+
+    def set_scanning_filter(self, **kwargs) -> None:
+        # Implement scanning filter setup
```

Comment:
Is this a TODO?

Reply:
Yes, I have not checked in detail what type of filters can you set with Bumble